src/templates/optimizing.js

```diff
@@ -928,6 +928,32 @@ function library() {
       }
       return $res;
     }
+
+    function invalidatePath(path) {
+        path.forEach((part, index) => {
+          triggerInvalidations(getAssignableObject(path, index), part, index === path.length - 1)
+        })
+    }
+
+    function set(path, value) {
+      ensurePath(path)
+      invalidatePath(path)
+      applySetter(getAssignableObject(path, path.length - 1), path[path.length - 1], value)
+    }
+
+    function splice(pathWithKey, len, ...newItems) {
+      ensurePath(pathWithKey)
+      const key = pathWithKey[pathWithKey.length - 1]
+      const path = pathWithKey.slice(0, pathWithKey.length - 1)
+      const arr = getAssignableObject(path, path.length)
+      const origLength = arr.length;
+      const end = len === newItems.length ? key + len : Math.max(origLength, origLength + newItems.length - len);
+      for (let i = key; i < end; i++ ) {
+        triggerInvalidations(arr, i, true);
+      }
+      invalidatePath(path)
```

Comment:
invalidatePath with the path without the key will cause the entire array to be invalidated

Reply:
Fixed. also added push setter

Reply:
Oh shit yea I forgot the invalidates bool flag.... sec

Reply:
It's fixed now.